File: prueba-tecnica-backend/src/main/java/com/lautaro_cenizo/entity/repuesto/RepuestoUsado.java
package com.lautaro_cenizo.entity.repuesto;

import com.lautaro_cenizo.entity.historial.HistorialServicio;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "RepuestosUsados")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RepuestoUsado {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    private Repuesto repuesto;

    @ManyToOne
    private HistorialServicio historialServicio;

    private Integer cantidad;
    private BigDecimal precioUnitario;

    @PrePersist
    public void prePersist() {
        this.precioUnitario = repuesto.getPrecio();
    }
}

File: prueba-tecnica-frontend/src/app/models/class/repuestoUsado.ts
import { HistorialServicio } from "./historialServicio";
import { Repuesto } from "./repuesto";

export class RepuestoUsado {
    id!: number;
    repuesto!: Repuesto;
    historialServicio!: HistorialServicio;
    cantidad!: number;
    precioUnitario!: number;
  
    constructor(data?: Partial<RepuestoUsado>) {
      Object.assign(this, data);
    }
  }